cpp/src/arrow/util/utf8.cc

```diff
@@ -73,10 +73,13 @@ ARROW_EXPORT void CheckUTF8Initialized() {
 
 }  // namespace internal
 
-static std::once_flag utf8_initialized;
+// static std::once_flag utf8_initialized;
 
 void InitializeUTF8() {
-  std::call_once(utf8_initialized, internal::InitializeLargeTable);
+  // std::call_once(utf8_initialized, internal::InitializeLargeTable);
+  // How should we do this? Maybe use
+  // https://en.wikibooks.org/wiki/More_C%2B%2B_Idioms/Nifty_Counter
+  internal::InitializeLargeTable();
```

Comment:
Note that you can refactor the ut8 stuff into a simple struct class that allocates the array on the heap. Then you use the same static + unique_ptr singleton technique and you'll get rid of the once_flag.